### ISSUE TITLE
Auto-generated index names don't match MySQL for composite keys

### ISSUE THREAD

**Issue by arvidfm** [dolthub/dolt#7960]
Consider two tables like:
```sql
CREATE TABLE main(
  id1 INT,
  id2 INT,
  id3 INT,
  PRIMARY KEY (id1, id2),
  KEY (id2, id3)
);

CREATE TABLE ref(
    main_id1 INT,
    main_id2 INT,
    main_id3 INT,
    FOREIGN KEY (main_id1, main_id2) REFERENCES main (id1, id2),
    CONSTRAINT `fk_ref_main` FOREIGN KEY (main_id2, main_id3) REFERENCES main (id2, id3)
);
```

In MySQL, this leads to two indexes being created on `ref` named as follows:
```sql
  KEY `main_id1` (`main_id1`,`main_id2`),
  KEY `fk_ref_main` (`main_id2`,`main_id3`),
```

In Dolt, however, they are named as:

```sql
  KEY `main_id1main_id2` (`main_id1`,`main_id2`),
  KEY `main_id2main_id3` (`main_id2`,`main_id3`),
```

This results in complications when writing migrations that manipulate indexes of old tables if compatibility with both Dolt and MySQL is required.

Complete runnable MWE that works with Percona Server 8.3, but causes an error with Dolt:
```sql
DROP TABLE IF EXISTS b_test, a_test;

CREATE TABLE a_test (id1 INT, id2 INT, id3 INT, PRIMARY KEY (id1, id2), KEY (id2, id3));

CREATE TABLE b_test(
    a_id1 INT,
    a_id2 INT,
    a_id3 INT,
    FOREIGN KEY (a_id1, a_id2) REFERENCES a_test (id1, id2),
    CONSTRAINT `fk_b_a` FOREIGN KEY (a_id2, a_id3) REFERENCES a_test (id2, id3)
);

ALTER TABLE b_test
    DROP FOREIGN KEY fk_b_a,
    DROP FOREIGN KEY b_test_ibfk_1,
    DROP KEY a_id1,
    DROP KEY fk_b_a;

DROP TABLE b_test, a_test;
```

**Comment by fulghum** [comment]
Thanks for reporting this behavior difference between Dolt and MySQL. I've got repro tests coded up and am starting to dig into the changes for this one.

**Comment by fulghum** [comment]
@arvidfm – The fix for this issue has been released in [Dolt 1.39.4](https://github.com/dolthub/dolt/releases/tag/v1.39.4). Thanks again for pointing out this behavior difference so we could fix it! 🙏  Let us know if you find any other issues and we'll be happy to help.
